### PR TITLE
Support provider > profile and profile per stage

### PR DESCRIPTION
**Pull request by rowanu**
***Implementing Issue:*** #1940

## What did you implement:

This PR has two improvements:

1. Actually load AWS Profiles properly - while there was code to set the profile name, it wasn't being recognised by the AWS SDK.
2. Add support for setting a profile via the `provider > profile` variable in `serverless.yml`.

In addition to this, I have documented how to support different profiles per stage.

This PR supersedes #1945.

## How did you implement it:

The profile is now set by loading the values from the `~/.aws/credentials` file [according to the docs](http://docs.aws.amazon.com/AWSJavaScriptSDK/guide/node-configuring.html) (see the heading "Using Profiles with the SDK").

The AWS provider plugin now attempts to load the profile specified in `provider > profile`, if no environment variable is set.

## How can we verify it:

* Add a profile name to `serverless.yml` in the `provider > profile` property, and run a `sls` command **without** setting an environment variable (e.g. `AWS_PROFILE`). The config-based property will be used.
* Add a `custom > profiles` configuration section and update your `provider > profile` property to use variables (like the example given in the updated docs for [provider account setup](./docs/guide/provider-account-setup.md)). Run an `sls` command specifying different stages and verify the that the correct profile is used.

## Todos:

- [x] Write tests
- [x] Fix linting errors
- [x] Provide verification config/commands/resources

### CODE REVIEW

**Comment by flomotlik** [comment]
I assume the conflict is in the docs. Can you move them to `docs/02-providers/aws/01-setup.md`. Just merged the new docs structure today (basically same file).

**Comment by flomotlik** [comment]
we don't have the `serverless.md` file any more, this should also be described as a default in the setup file mentioned before. Happy to help if you have questions about this.

**Comment by rowanu** [comment]
Docs tweaked, let me know if anything else required.

**Comment by flomotlik** [comment]
Great change, will also review in-depth on Thursday!

**Comment by WooDzu** [comment]
Hi @rowanu I tested your change and it works as described. AWS_PROFILE not needed any more!

My config is:
```yml
provider:
  name: aws
  runtime: nodejs4.3
  memorySize: 128
  stage_default: dev
  stage: ${opt:stage, self:provider.stage_default}
  profile: ${self:provider.stage}
  region: eu-central-1
```

Btw. is there any reason why `${opt:stage, dev}` or `${opt:stage, "dev"}` does not work and requires additional config entry?

One suggestion I have is to explain what is exactly "a stage" and how it reflects on aws, but I suppose this is more of a general issue.

**Comment by flomotlik** [comment]
>Btw. is there any reason why ${opt:stage, dev} or ${opt:stage, "dev"} does not work and requires additional config entry?

We haven't implemented it yet, but discussed it during the implementation. We wanted to wait first if its something that a lot of people need and then implement it as a followup.

**Comment by leonardoalifraco** [comment]
👍

**Comment by andymac4182** [comment]
Anything we can to do to help this get merged?

**Comment by flomotlik** [comment]
@andymac4182 extending the day to 36 hours :D

Had to do some Serverless internal stuff for the last 2 days, but tomorrow is all freed up for diving back into the PR's and issues. Thanks guys for pushing this hard, really appreciate it and makes sure we move faster!

**Comment by konstantinvlasenko** [comment]
Hi All,
Sorry for inconvenience :( But I am lost about this profile configuration.
How can I specify not a default one from ~/.aws/credentials?

**Comment by andymac4182** [comment]
@konstantinvlasenko This Pull Request is what enables you to specify the correct profile to use. @WooDzu has an example of how to do that above after this.

File: docs/02-providers/aws/01-setup.md [hunk 1]
```diff
@@ -62,4 +66,42 @@ Default region name [None]: us-west-2
 Default output format [None]: ENTER
 ```
 
-You can also edit the credentials file which is located in `~/.aws/credentials` directly. Read more about that file in the [AWS documentation](http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-config-files)
+Credentials are stored in INI format in `~/.aws/credentials`, which you can edit directly if needed. Read more about that file in the [AWS documentation](http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-config-files)
+
+You can even set up a different profiles for different accounts, which can be used by Serverless as well. To specify a default profile to use, you can add a `profile` setting to your `provider` configuration in `serverless.yml`:
```
Comment (flomotlik) [inline docs/02-providers/aws/01-setup.md]: "a different profiles" -> "different profiles"

**Pull request merged**